pkg/storage/ingest/pusher.go

```diff
@@ -460,12 +466,13 @@ func (p *parallelStorageShards) run(queue *batchingQueue) {
 		p.metrics.batchAge.Observe(time.Since(wr.startedAt).Seconds())
 		p.metrics.timeSeriesPerFlush.Observe(float64(len(wr.Timeseries)))
 		processingStart := time.Now()
+		requestContents := requestContents(wr.WriteRequest)
 
```

Comment:
What if we do `wr.WriteRequest = nil` here? Given it will be unsafe to use, better a panic rather than data corruption, if in the future we'll try to read it.

Reply:
Same in `pusherConsumer.Consume()`.

Reply:
I cleared it at the spot where it's freed (it feels like a logical extension of freeing). The rename makes it clear at the call site too. Is that enough? :thinking: